src/components/results/result.js

```diff
@@ -11,6 +11,27 @@ class Result extends React.Component {
     return value.join(", ");
   }
 
+
+  // Highlight the search words in the result text.
+  HighlightWords(field, text){
+    if (field.type === 'text' && field.value !== undefined){
+      // replace "&", "|", "(", ")", "*", "'", """ and duplicated whitespace with only one whitespace
+      let values = field.value.replaceAll(/\&|\*|\(|\)|\||\"|\'/g,"").replaceAll(/\s+/g," ").split(" ");
+
+      // split original text by search words insensitively by regex pattern
+      // g is for regex global and i is for insensitive.
+      values.map((value) => {
+        let regex = new RegExp(value, "gi");
+        text = text.split(regex).join("<span style='background-color:yellow;'>" + value + "</span>");
+      })
+
```

Comment:
Replace search words in the text with the yellow background color. We could also use `replace` with a regex insensitive pattern.